pkg/types/ibmcloud/platform.go

```diff
@@ -0,0 +1,61 @@
+package ibmcloud
+
+// Platform stores all the global configuration that all machinesets use.
+type Platform struct {
+	// Region specifies the IBM Cloud region where the cluster will be
+	// created.
+	Region string `json:"region"`
+
+	// CISInstanceCRN is the Cloud Internet Services CRN of the base domain DNS
+	// zone.
+	CISInstanceCRN string `json:"cisInstanceCRN"`
```

Comment:
Should this be optional? Can we deduce this value from the base domain field?

Reply:
Unfortunately not. This field is required. A user could technically have multiple separate instances of CIS managing the same DNS zone. It's a little bit of a weird and rare state, but something we have to handle nonetheless.

Reply:
If the user had multiple instances of CIS managing the same DNS zone, then they could use this field. But I don't understand the leap from there to requiring that *every* user specify this.

Reply:
And, to be clear, I am not suggesting that we remove the field. I am suggesting that the field be optional for the user to specify rather than mandatory.

Reply:
That's a very good point. Makes a lot of sense. I'll put together some changes to support it ASAP

Reply:
@staebler This got me thinking more and I discovered the following little tidbit of truth. Though it is possible
to manage a single DNS zone with multiple CIS instances, only one zone can be in the "Active" state at a time. As a result, we know which CIS instance to use based off of its zone's state. For that reason, I added a commit to remove the `cisInstanceCRN` field entirely.

Reply:
We may want to get the CRN from the `baseDomain` and save it to various metadata for Terraform to pickup, but we'll cross that bridge in the future PRs.

Reply:
Yep, we can save the CRN elsewhere if need be.